frontends/common/constantParsing.h

```diff
@@ -17,11 +17,71 @@ limitations under the License.
 #ifndef _FRONTENDS_COMMON_CONSTANTPARSING_H_
 #define _FRONTENDS_COMMON_CONSTANTPARSING_H_
 
-#include <gmpxx.h>
-#include "ir/ir.h"
-#include "lib/source_file.h"
+#include <boost/optional.hpp>
+#include "lib/cstring.h"
 
-// helper for parsing constants
-IR::Constant* cvtCst(Util::SourceInfo srcInfo, const char* s, unsigned skip, unsigned base);
+namespace IR {
+class Constant;
+}  // namespace IR
+
+namespace Util {
+class SourceInfo;
+}  // namespace Util
+
+/**
+ * An unparsed numeric constant. We produce these as token values during
+ * lexing. The parser is responsible for actually interpreting the raw text as a
+ * numeric value and transforming it into an IR::Constant using parseConstant().
+ *
+ * To illustrate how a numeric constant is represented using this struct,
+ * here is a breakdown of '16w0x12':
+ *
+ *          ___
+ *         /                                    ___
+ *         |                                   /
+ *         |   bitwidth (if @hasWidth)        |       16
```

Comment:
It would've taken a lot fewer if I turned this diagram "sideways", but then I didn't have enough horizontal space. =\